### DISCUSSION TITLE
训练中英文识别模型时怎样定义max_text_length

### DISCUSSION THREAD

**Discussion by ericosmic** [PaddlePaddle/PaddleOCR#12924] [Q&A] [answered]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：ubuntu
- 版本号/Version：Paddle：2.3  PaddleOCR：2.7 问题相关组件/Related components： tools/train.py 
- 运行指令/Command Code：```
python3 -m paddle.distributed.launch --log_dir=./debug/ --gpus '2,3'  tools/train.py \
    -c configs/rec/PP-OCRv4/ch_PP-OCRv4_rec.yml
```
- 完整报错/Complete Error Message：

在训练中英文通用识别模型时，怎样确定max_text_length ， 英文长度是按英文字母计算？ 还是按英文单词计算？ 需要将空格计算在内吗？

我们提供了AceIssueSolver来帮助你解答问题，你是否想要它来解答(请填写yes/no)?/We provide AceIssueSolver to solve issues, do you want it? (Please write yes/no):

请尽量不要包含图片在问题中/Please try to not include the image in the issue.

**Comment by zz1988** [comment, marked as answer]
英文长度是按照字母计算的，需要将空格计算在内。
我印象中，定义的`max_text_length`需要比实际的长度至少多两位数。
比如要想识别身份证号，实际中身份证号是18位，那在config文件中定义的`max_text_length`至少要为20，否则会报错。
